nntrainer/tensor/tensor.cpp

```diff
@@ -750,6 +752,49 @@ Tensor Tensor::getSharedDataTensor(const TensorDim dim_, unsigned int offset,
   return ret;
 }
 
+std::vector<Tensor> Tensor::split(unsigned num_size, int axis) {
+  if (axis == -1) {
+    axis = 3;
+  }
+  NNTR_THROW_IF(!(0 <= axis && axis < 4), std::invalid_argument)
+    << "cannot split axis of axis: " << axis;
+
+  NNTR_THROW_IF(dim.getTensorDim(axis) % num_size != 0, std::invalid_argument)
+    << "axis is not divisible by num_size, axis: " << axis
+    << " num size: " << num_size;
+
+  auto ret_dim = dim;
+  auto new_dim = dim.getTensorDim(axis) / num_size;
+  ret_dim.setTensorDim(axis, new_dim);
+
+  auto iter_value = [this, &ret_dim](std::array<unsigned, 4> &loc) {
```

Comment:
this return 1 value at a time with this operation, using getValue() - this is needed only along the axis for the split right and not along all the dimensions?

Reply:
yeah. That can be optimized.